File: integrasjon/src/test/java/no/nav/melosys/integrasjon/felles/MDCOperationsTest.java
package no.nav.melosys.integrasjon.felles;

import no.nav.melosys.integrasjon.felles.mdc.MDCOperations;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MDCOperationsTest {

    @Test
    public void test_generateCallId() {

        String callId1 = MDCOperations.generateCallId();

        assertThat(callId1).isNotNull();

        String callId2 = MDCOperations.generateCallId();

        assertThat(callId2).isNotNull().isNotEqualTo(callId1);
    }

    @Test
    public void test_mdc() {
        MDCOperations.putToMDC("myKey", "myValue");

        assertThat(MDCOperations.getFromMDC("myKey")).isEqualTo("myValue");

        MDCOperations.remove("myKey");

        assertThat(MDCOperations.getFromMDC("myKey")).isNull();
    }
}

File: integrasjon/src/test/kotlin/no/nav/melosys/integrasjon/felles/mdc/MDCOperationsTest.kt
package no.nav.melosys.integrasjon.felles.mdc

import io.kotest.matchers.nulls.shouldBeNull
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.string.shouldBeEqualIgnoringCase
import io.kotest.matchers.string.shouldNotBeEqualIgnoringCase
import no.nav.melosys.integrasjon.felles.mdc.MDCOperations.Companion.generateCallId
import no.nav.melosys.integrasjon.felles.mdc.MDCOperations.Companion.getFromMDC
import no.nav.melosys.integrasjon.felles.mdc.MDCOperations.Companion.putToMDC
import no.nav.melosys.integrasjon.felles.mdc.MDCOperations.Companion.remove
import org.junit.jupiter.api.Test

class MDCOperationsTest {

    @Test
    fun test_generateCallId() {
        val callId1 = generateCallId()
        callId1.shouldNotBeNull()
        val callId2 = generateCallId()
        callId2.shouldNotBeNull()
            .shouldNotBeEqualIgnoringCase(callId1)
    }

    @Test
    fun test_mdc() {
        putToMDC("myKey", "myValue")
        getFromMDC("myKey")
            .shouldNotBeNull()
            .shouldBeEqualIgnoringCase("myValue")
        remove("myKey")
        getFromMDC("myKey").shouldBeNull()
    }

}
